kbase-extension/static/kbase/js/common/cellComponents/tabs/jobStatus/jobStateListRow.js

```diff
@@ -1,188 +1,292 @@
+            const $currentRow = $(e.target).closest('tr');
+            const $dtTable = $(e.target).closest('table').DataTable();
+            const dtRow = $dtTable.row($currentRow);
+
+            if (dtRow.child.isShown()) {
+                // This row is already open - close it
+                dtRow.child.hide();
+                $currentRow.removeClass('vertical_collapse--open');
+                if (widgets.log) {
+                    widgets.log.stop();
+                }
+                dtRow.child.remove();
+                return;
+            }
+
+            // create the child row contents, add to the child row, and show it
+            const str = div({
+                class: `${cssBaseClass}__log_container`,
+                dataElement: 'job-log-container',
+            });
+            dtRow.child(false);
+            dtRow.child(str).show();
+
+            // add the log widget to the next `tr` element
+            widgets.log = JobLogViewer.make({ showHistory: true });
+            widgets.log.start({
+                node: $currentRow.next().find('[data-element="job-log-container"]')[0],
+                jobId: jobId,
+                jobState: jobState,
+            });
+            $currentRow.addClass('vertical_collapse--open');
         }
 
-        function start(arg) {
+        /**
+         * Start the job status row widget
+         * @param {object} args, with keys
+         *      jobState:  a job state object for the job to be represented
+         *      name: the value of the input parameter(s) for the job
+         *      node: the node to insert the row into (will presumably be a `tr` element)
+         */
+        function start(args) {
             return Promise.try(() => {
-                container = arg.node;
-                updateRowStatus(arg.initialState, arg.name);
-                events.attachEvents(container);
+                const requiredArgs = ['jobState', 'name', 'node'];
+                if (
+                    Object.prototype.toString.call(args) !== '[object Object]' ||
+                    !requiredArgs.every((arg) => arg in args && args[arg])
+                ) {
+                    throw new Error(
+                        'invalid arguments supplied' +
+                            '\n\n' +
+                            'you supplied ' +
+                            JSON.stringify(args)
+                    );
+                }
+
+                if (!Jobs.isValidJobStateObject(args.jobState)) {
+                    throw new Error('invalid job object supplied');
+                }
+                container = args.node;
+                ui = UI.make({ node: container });
+                container.innerHTML = buildRow(args.name);
+                container.onclick = (e) => {
+                    selectRow(e);
+                    showHideChildRow(e);
+                };
+                jobId = args.jobState.job_id;
+                _updateRowStatus(args.jobState);
```

Comment:
You could also just pass this to `updateState` and let it handle the errors from line 167.

Reply:
updateState requires the jobId to be set, and before doing that, you should check that the job object is valid. I've reordered the lines to make that a bit more obvious.